src/timeline/RoomlistModel.cpp

```diff
@@ -1081,6 +1081,13 @@ FilteredRoomlistModel::toggleTag(QString roomid, QString tag, bool on)
     }
 }
 
+void
+FilteredRoomlistModel::copyLink(QString roomid)
+{
+    QString link = QString("https://matrix.to/#/%1").arg(roomid);
```

Comment:
I think it makes more sense to reuse the copyLinkToEvent function: https://github.com/Nheko-Reborn/nheko/blob/master/src/timeline/TimelineModel.cpp#L1756

You should be able to refactor (and rename) it a bit, so that it can be used for both, creating room links and event links. This is important since:

- Roomlinks should have via's to tell other clients how to find the room.
- You should urlescape all parameters, since those could be malicious
- You should use the room alias if possible and fallback to the roomid, if the alias is not set

Thank you for working on this though!

Reply:
Thanks for the heads up. I split copyLinkToEvent into two separate functions.